classes/village/VillageBuildingConfigData.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+require_once __DIR__ . "/../war/WarManager.php";
+
+class VillageBuildingConfigData {
+        public function __construct(
+        private string $key,
+        private string $name,
+        private string $description,
+        private string $phrase,
+        private string $background_image,
+        private array $max_healths,
+        private array $construction_costs = [],
+        private array $construction_times = [],
+        private array $upgrade_sets = [],
+    ) {}
+    public function getKey(): string {
+        return $this->key;
+    }
+    public function getName(): string {
+        return $this->name;
+    }
+    public function getDescription(): string {
+        return $this->description;
+    }
+    public function getPhrase(): string {
+        return $this->phrase;
+    }
+    public function getBackgroundImage(): string {
+        return $this->background_image;
+    }
+    public function getMaxHealth($tier): int {
+        return $this->max_healths[$tier] ?? 0;
+    }
+    public function getConstructionCostMaterials($tier): int {
+        return $this->construction_costs[$tier][WarManager::RESOURCE_MATERIALS] ?? 0;
```

Comment:
This and similar functions should probably throw a RuntimeException if its an invalid tier so we don't inadvertently allow free construction if we messed up the tier we passed into this check